autoscheduler/scraper/tests/api_tests.py

```diff
@@ -0,0 +1,253 @@
+                    current_enrollment=40, instructor=test_instructors[0]),
+            Section(crn=12346, id='000002', subject='CSCE', course_num='310',
+                    section_num='502', term_code='201931', min_credits='3',
+                    honors_only=False, web_only=False, max_enrollment=50,
+                    current_enrollment=40, instructor=test_instructors[1],
+                    instructor_gpa=3.2),
+        ]
+        self.meetings = [
+            Meeting(id='0000010', meeting_days=[True] * 7, start_time=time(11, 30),
+                    end_time=time(12, 20), meeting_type='LEC', section=self.sections[0]),
+            Meeting(id='0000011', meeting_days=[True] * 7, start_time=time(9, 10),
+                    end_time=time(10), meeting_type='LEC', section=self.sections[0]),
+            Meeting(id='0000020', meeting_days=[True] * 7, start_time=time(11, 30),
+                    end_time=time(12, 20), meeting_type='LEC', section=self.sections[1]),
+            Meeting(id='0000021', meeting_days=[False] * 7, start_time=time(9, 10),
+                    end_time=time(10), meeting_type='LAB', section=self.sections[1]),
+        ]
+        for course in self.courses:
+            course.save()
+        for section in self.sections:
+            section.save()
+        for meeting in self.meetings:
+            meeting.save()
+
+    def test_api_terms_displays_all_terms(self):
+        """ Tests that /api/terms returns ordered list of all terms in database """
+        # Arrange
+        expected = {
+            '201831': 'Fall 2018 - College Station',
+            '201931': 'Fall 2019 - College Station',
+            '202031': 'Fall 2020 - College Station',
+        }
+
+        # Act
+        response = self.client.get('/api/terms')
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(expected, response)
+
+    def test_api_course_serializer_gives_expected_output(self):
+        """ Tests that the course serializer yields the correct data """
+        # Arrange
+        expected = {'title': 'Introduction to Computing', 'credit_hours': 3}
+
+        # Act
+        serializer = CourseSerializer(self.courses[0])
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_course_serializer_handles_null_credit_hours(self):
+        """ Tests that the course serializer correctly handles null credit_hours """
+        # Arrange
+        expected = {'title': 'Sports Law', 'credit_hours': None}
+
+        # Act
+        serializer = CourseSerializer(self.courses[4])
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_course_gives_valid_response_csce(self):
+        """ Tests that /api/course?dept=CSCE&course_num=181&term=201931 gives the
+            correct output
+        """
+        # Arrange
+        expected = CourseSerializer(self.courses[0])
```

Comment:
If the `CourseSerializer` is wrong but the `api/course` route is correct, this test will fail (when it shouldn't), so we should probably change the expected to be hard-coded. Could also just make a helper function that takes in a course and returns the necessary fields.